packages/migrations/src/migrateDockerNetworkIpRange/ensureDockerNetworkConfig/recreateDockerNetwork.ts

```diff
@@ -0,0 +1,71 @@
+import {
+  disconnectAllContainersFromNetwork,
+  docker,
+} from "@dappnode/dockerapi";
+import { logs } from "@dappnode/logger";
+import Dockerode from "dockerode";
+import { isEmpty } from "lodash-es";
+
+/**
+ * Recreates the docker network with the configuration defined in networkOptions:
+ *
+ * 1. Disconnect al docker containers from a docker network
+ * 2. Remove the docker network
+ * 3. Creates again the docker network with the config
+ *
+ * @param networkToRemove dockerode network to remove
+ * @param newNetworkOptions dockerorde create options object
+ */
+export async function recreateDockerNetwork(
+  networkToRemove: Dockerode.Network,
+  newNetworkOptions: Dockerode.NetworkCreateOptions
+): Promise<Dockerode.Network> {
+  logs.info(`disconnecting all containers from ${networkToRemove.id}`);
+  await disconnectAllContainersFromNetwork(networkToRemove).catch((e) =>
+    logs.error(
+      `error while disconnecting all containers from the network: ${e.message}`
+    )
+  );
+
+  await removeConnectedContainersIfAny(networkToRemove).catch((e) =>
```

Comment:
I would retry to disconnect the containers before removing them

Reply:
this is actually the flow, isnt it?

Reply:
I don't think so. We try to disconnect them and then we directly remove them. Removing those containers should be the last resort as the container might be the `web3signer` for example. I would first retry disconnecting the containers for some attempts

Reply:
So it is, the flow:
1. Disconnect them
2. If still some docker containers connected, then remove them

As you said we could do a removal retry if there are still some connected containers but, how many times, during how much time?

Reply:
I would retry like 5 times or so. Leaving a delay of some seconds

Reply:
Do we know what happens if we try to disconnect/connect a restarting container from the network?